### PR TITLE
Rose

### PR DESCRIPTION
**Pull request by namandistro**
Using this template will help us better understand your Instant Answer and assist you when necessary. Simply copy and paste the markdown below into the description of your GitHub pull request and complete as appropriate.

What does your Instant Answer do?

Display a Cheat sheet for Different roses and their meanings  
What is the data source for your Instant Answer? (Provide a link if possible)

http://rkdn.org/roses/colors.asp

Why did you choose this data source?

Complete and fairly trusted.

Are there any other alternative (better) data sources?

No.

Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)

everybody.

Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?

Yes, It was a suggestion zac in a email
Are you having any problems? Do you need our help with anything?

No.

What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)
screenshot from 2015-07-29 12 57 13

-----
IA Page: https://duck.co/ia/view/rose_meanings_cheat_sheet

### CODE REVIEW

File: share/goodie/cheat_sheets/rose.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
+{
+    "name": "Rose Cheat Sheet",
+    "id": "rose_cheat_sheet",
+    "description": "Not all rose are grown naturally, Some are even artificial.",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/rose.json]: This shouldn't have a capital `s` on `some`

File: share/goodie/cheat_sheets/rose.json [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            "val": "A rose of any color depicts utmost devotion",
+            "key": "Single"
+        },  {
+            "val": "Roses entwined together communicate 'Marry me'",
+            "key": "Two"
+        },  {
+            "val": "Roses signify a need to be loved or cherished",
+            "key": "Six"
+        },  {
+            "val": "roses assure the recipient they are truly and deeply loved",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/rose.json]: Can you make all of them `roses` ? I think it looks better, however @abeyang may think otherwise!

**Comment by namandistro** [comment]
@mintsoft Iz their any thing else i need to change?

**Comment by mintsoft** [comment]
@abeyang Any thoughts?

File: share/goodie/cheat_sheets/rose.json [hunk 3]
```diff
@@ -0,0 +1,102 @@
+{
+    "name": "Rose Cheat Sheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/rose.json]: Let's remove "Cheat Sheet" from the name.

File: share/goodie/cheat_sheets/rose.json [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    "id": "rose_cheat_sheet",
+    "description": "Not all rose are grown naturally, some are even artificial.",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/rose.json]: I think this should really describe the actual cheat sheet. Explaining that these are the common meanings for each rose petal colour

File: share/goodie/cheat_sheets/rose.json (hunk 3 above)
Comment (moollaza) [inline share/goodie/cheat_sheets/rose.json]: Actually lets change the name to "Rose Color Meanings" or similar

File: share/goodie/cheat_sheets/rose.json [hunk 5]
```diff
@@ -0,0 +1,102 @@
+{
+    "name": "Rose Cheat Sheet",
+    "id": "rose_cheat_sheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/rose.json]: please change to `rose_meanings_cheat_sheet`

File: share/goodie/cheat_sheets/rose.json [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        "sourceName": "Morrison Garden",
+        "sourceUrl": "http://rkdn.org/roses/colors.asp"
+    },
+    "section_order": [
+                        "Colour of Rose",
+                        "Number of Rose"
+                     ],
+    "sections": {
+        "Colour of Rose": [{
+            "val": "Love, Beauty, Courage and Respect, Romantic Love, Congratulations, I Love You, Job Well Done, Sincere Love, Respect, Courage & Passion",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/rose.json]: Let's stick with "&" or "and". The latter is probably better

File: share/goodie/cheat_sheets/rose.json [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+            "val": "Unconscious Beauty",
+            "key": "Dark Red"
+        }, {
+            "val": "I Love You",
+            "key": "Single Red"
+        }, {
+            "val": "Unconscious Beauty",
+            "key": "Burgundy"
+        }, {
+            "val": "Purity, Innocence, Silence, Secrecy, Reverence, Humility, Youthfulness, I am worthy of you, Heavenly ",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/rose.json]: Let's remove the trailing space here and in the others

File: share/goodie/cheat_sheets/rose.json [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+            "key": "Lavender"
+        }, {
+            "val": "Death, Farewell",
+            "key": "Black"
+        }, {
+            "val": "The unattainable, the impossible",
+            "key": "Blue"
+        }, {
+            "val": "Symbolic of purity and loveliness",
+            "key": "[Red] [Rosebud]"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/rose.json]: I don't think we need to separate these. "Red Rosebud" should be fine.

File: share/goodie/cheat_sheets/rose.json [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        }, {
+            "val": "Symbolic of girlhood",
+            "key": "[White] [Rosebud]"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/rose.json]: Same here, I don't think we should wrap these in `[]`

File: share/goodie/cheat_sheets/rose.json [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+        }, {
+            "val": "Love at first sight",
+            "key": " [Thorn-less] [Rose]"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/rose.json]: Same here, also let's remove the leading space.

File: share/goodie/cheat_sheets/rose.json [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+        }],
+        "Number of Rose": [
```
Comment (moollaza) [inline share/goodie/cheat_sheets/rose.json]: Typo: "Number of rose**s**"

File: share/goodie/cheat_sheets/rose.json [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+            {
+            "val": "A rose of any color depicts utmost devotion",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/rose.json]: Let's make sure these are proper sentences and have sentence casing.

File: share/goodie/cheat_sheets/rose.json [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+            "key": "Single"
+        },  {
+            "val": "roses entwined together communicate 'Marry me'",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/rose.json]: `When entwined, two roses communicate "Marry Me"`

File: share/goodie/cheat_sheets/rose.json [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+            "key": "Two"
+        },  {
+            "val": "roses signify a need to be loved or cherished",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/rose.json]: `"Signify a need to be loved or cherisdhed"`

File: share/goodie/cheat_sheets/rose.json [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+            "key": "Six"
+        },  {
+            "val": "roses assure the recipient they are truly and deeply loved",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/rose.json]: "Assure the recipient... "

File: share/goodie/cheat_sheets/rose.json [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+            "key": "Eleven"
+        },  {
+            "val": "roses indicate a secret admirer",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/rose.json]: "Indicates a secret admireer"

**Comment by namandistro** [comment]
@moollaza Bro, So much emails at once.... Just Kidding.... I will make all the changes as recommended, Thanks alot!

File: share/goodie/cheat_sheets/rose.json (hunk 6 above)
Comment (namandistro) [inline share/goodie/cheat_sheets/rose.json]: @moollaza can you please explain the change better

File: share/goodie/cheat_sheets/rose.json (hunk 7 above)
Comment (namandistro) [inline share/goodie/cheat_sheets/rose.json]: @moollaza can you please explain the change better

File: share/goodie/cheat_sheets/rose.json (hunk 10 above)
Comment (namandistro) [inline share/goodie/cheat_sheets/rose.json]: @moollaza can you please explain the change better

**Comment by moollaza** [comment]
:+1:

**Comment by namandistro** [comment]
@Moollaza I had made the recommended changes as you can see....!

**Comment by abeyang** [comment]
Based on the screenshot (which was a few updates ago, I know), just make sure there's proper title casing (eg, first letter should be capitalized). Other than that, looks good -- and this would make great use of our **reference sheet**, once we release that in v2!

**Comment by namandistro** [comment]
@abeyang I made all the recommended changes. just have a look.

**Comment by abeyang** [comment]
@namandistro Hey, @zachthompson and I both tried this on our dev machines, and it's throwing an error. Can you double-check?

File: share/goodie/cheat_sheets/rose.json [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+        }, {
+            "val": "Love at first sight",
+            "key": "Thorn-less Rose"
+        }],
+        "Number of Roses": [
+            {
+            "val": "Rose of any color depicts utmost devotion",
+            "key": "Single"
+        },  {
+            "val": "When entwined, two roses communicate "Marry Me"",
```
Comment (zachthompson) [inline share/goodie/cheat_sheets/rose.json]: Need to escape the double quotes around `Marry Me`.

**Comment by namandistro** [comment]
@abeyang now have a look at it... and let me know...

**Comment by moollaza** [comment]
@namandistro LGTM :+1: 

Thanks a lot!

**Comment by namandistro** [comment]
@moollaza Code iz live but no notification or credit!

**Comment by moollaza** [comment]
@namandistro sorry, I'm not sure how but it looks like the ia page was deleted. I've create a new one and set you as the contributor. It should go live with the next deploy on Monday/Tuesday.

**Pull request merged**